reana/reana_dev/run.py

```diff
@@ -279,9 +390,24 @@ def run_ci(
     help="Additional operational options for the workflow execution. "
     "E.g. CACHE=off.",
 )
+@click.option(
+    "--submit-only", is_flag=True, help="Do not wait for workflows to finish."
+)
+@click.option(
+    "--check-only", is_flag=True, help="Wait for previously submitted workflows."
```

Comment:
When `--check-only` is used, it would be good not to loop over statuses, as the PR does now:

```console
$ reana-dev run-example -c DEMO -b kubernetes -b htcondorcern -b slurmcern --check-only 
[2021-03-16T19:30:02] : Running the following matrix:
Demos: reana-demo-root6-roofit,reana-demo-atlas-recast,reana-demo-helloworld,reana-demo-worldpopulation
Workflow engines: cwl,serial,yadage
Compute backends: htcondorcern,slurmcern,kubernetes
[2021-03-16T19:30:07] reana-demo-root6-roofit: reana-client status -w root6-roofit-cwl-htcondorcern
NAME                            RUN_NUMBER   CREATED               STARTED               STATUS    PROGRESS
root6-roofit-cwl-htcondorcern   1            2021-03-16T18:27:58   2021-03-16T18:28:00   running   0/2     
[2021-03-16T19:30:13] reana-demo-root6-roofit: reana-client status -w root6-roofit-cwl-htcondorcern
NAME                            RUN_NUMBER   CREATED               STARTED               STATUS    PROGRESS
root6-roofit-cwl-htcondorcern   1            2021-03-16T18:27:58   2021-03-16T18:28:00   running   0/2     
...
```

but rather print the current status, so that the person running the test suite could take a "snapshot" of the current state.

IOW, in the sequential mode, we want to wait until the end of the loop for each job, and move on to the next job; but in the parallel test mode, we are interested in seeing the snapshot at some moment in time of everything that is running, without waiting for anything, and then to retake that same snapshot 15 minutes later, 30 minutes later, etc.

Can you amend not to do status sleep while doing for the `--check-only` view of the results?

Reply:
BTW, this also means not to stop after failures:

```console
$ reana-dev run-example -c DEMO -b kubernetes -b htcondorcern -b slurmcern --check-only
...
[2021-03-16T20:00:28] reana-demo-root6-roofit: Command 'reana-client logs -w root6-roofit-cwl-htcondorcern | grep -c 'RooChebychev::bkg'' returned non-zero exit status 1.
```

but rather continue making the "snapshot" of the test cases, and then finish with some mini-report like:

```console
[2021-03-16T20:02:27] reana-dev run-example demo summary:
- 29 demo workflows run
- 27 passed
- 2 failed (foo-cwl-slurmcern, bar-yadage-kubernetes)
```

Reply:
P.S. Also, can you please enrich `CHANGES.rst` with a one-liner about this improvement?

Reply:
@tiborsimko It should be ready now, the improvement for `--check-only` is documented in https://github.com/reanahub/reana/issues/489, thanks for the detailed description 🙏